asserts/constraint.go

```diff
@@ -263,6 +275,38 @@ func (matcher evalAttrMatcher) match(apath string, v interface{}, ctx *attrMatch
 	return nil
 }
 
+type refAttrMatcher struct {
+	// supports $PLUG_PUBLISHER_ID
```

Comment:
we probably want to support SLOT as well to avoid confusing incosinsistency?

Reply:
I've added this now along with new tests